protocol/tx_test.go

```diff
@@ -1,4 +1,4 @@
-package protocol
```

Comment:
I admit I would not have expected things like moving Snapshot to another package to fall out of rewriting package validation in terms of bc.Y instead of bc.X. Maybe we should pull on this thread a little more to understand the ultimate motivation for it?